panel/widgets/tree.py

```diff
@@ -0,0 +1,238 @@
+        Size of the icons and text in pixels.""")
+
+    select_multiple = param.Boolean(default=True, doc="""
+        Whether multiple nodes can be selected or not""")
+
+    plugins = param.List(doc="""
+        Configure which additional plugins will be active.  Should be
+        an array of strings, where each element is a plugin name that
+        are passed to jsTree.""")
+
+    sort = param.Boolean(default=False, doc="""
+        Whether to sort nodes alphabetically.""")
+
+    show_icons = param.Boolean(default=True, doc="Whether to use icons or not.")
+
+    show_dots = param.Boolean(default=False, doc="""
+        Whether to show dots on the left as part of the tree.""")
+
+    show_stripes = param.Boolean(default=False, doc="""
+        Whether to show stripes on alternating rows.""")
+
+    value = param.List(default=[], doc="""
+        List of currently selected leaves and nodes""")
+
+    _new_nodes = param.List(default=None, doc="Children to push to tree")
+
+    _rename: ClassVar[Mapping[str, str | None]] = {
+        "icon_size": None,
+        "select_multiple": "multiple",
+        "name": None,
+        "value": "checked",
+    }
+
+    def __init__(self, **params):
+        click_handler = params.pop('on_click', None)
+        self._click__handlers = []
+        super().__init__(**params)
+        if click_handler:
+            self.on_click(click_handler)
+
+    def _on__click(self, event: NodeEvent):
+        """
+        Method called when a node is clicked.
+        """
+
+    def on_click(
+        self, callback: Callable[[param.parameterized.Event], None | Awaitable[None]]
+    ) -> param.parameterized.Watcher:
+        """
+        Register a callback to be executed when the `Button` is clicked.
+
+        The callback is given an `Event` argument declaring the number of clicks
+
+        Example
+        -------
+
+        >>> button = pn.widgets.Button(name='Click me')
+        >>> def handle_click(event):
+        ...    print("I was clicked!")
+        >>> button.on_click(handle_click)
+
+        Arguments
+        ---------
+        callback:
+            The function to run on click events. Must accept a positional `Event` argument. Can
+            be a sync or async function
+
+        Returns
+        -------
+        watcher: param.Parameterized.Watcher
+          A `Watcher` that executes the callback when the button is clicked.
+        """
+        self._click__handlers.append(callback)
+
+    def _to_json(
+        self, id_, label, parent: str = None, children: Optional[list] = None,
+        icon: str = None, state: dict[str, bool] = {}, **kwargs
+    ):
+        if "selected" not in state:
+            state["selected"] = id_ in self.value
+        jsn = dict(id=id_, text=label, children=children or [], state=state, **kwargs)
+        if parent:
+            jsn["parent"] = parent
+        if icon:
+            jsn["icon"] = icon
+        else:
+            jsn["icon"] = "jstree-leaf"
```

Comment:
Is it possible to let people specify any icon from [tabler-icons.io](https://tabler-icons.io/) from other places? This was always something I wanted to try get working but never got around to. It would especially be useful in a generic Tree when people are using it for all sorts of stuff. Having DNA icons for taxonomy things for my genome projects I am working on, or using local custom icons for a branching history tree, or any other tree.